### PR TITLE
[OTE] Add support for second opinions

### PR DESCRIPTION
**Pull request by loldot**
- [x] New feature
- [ ] Bug fix
- [ ] High impact

**Description of work:**
AB#28735
Add endpoints for requesting second opinions from outside competence centres

**Testing:**
- [x] Can be tested
- [x] Automatic tests created / updated
- [x] Local tests are passing

**Checklist:**
- [x] Considered automated tests
- [x] Considered updating specification / documentation
- [x] Considered work items 
- [x] Considered security
- [x] Performed developer testing
- [x] Checklist finalized / ready for review

### CODE REVIEW

File: src/backend/api/Fusion.Resources.Domain/Commands/SecondOpinions/GetSecondOpinions.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                }
+
+                if (request.RequestId.HasValue)
+                {
+                    query = query.Where(x => x.RequestId == request.RequestId);
+                }
+
+                if(request.Query.HasFilter)
+                {
+                    var filter = request.Query.Filter.GetFilterForField("state");
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Domain/Commands/SecondOpinions/GetSecondOpinions.cs:112]: Can't you just filter on what you pass along ?
Also. Have you seen the extension => query.ApplyOdataFilters(xxxxxxx ?

File: src/backend/api/Fusion.Resources.Domain/Commands/SecondOpinions/AddSecondOpinion.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Fusion.Resources.Domain.Commands
+{
+    public class AddSecondOpinion : TrackableRequest<QuerySecondOpinion>
+    {
+        private string title;
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Domain/Commands/SecondOpinions/AddSecondOpinion.cs:18]: Can probably make the variables readonly

File: src/backend/api/Fusion.Resources.Domain/Models/SecondOpinions/QuerySecondOpinion.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        {
+            Id = secondOpinion.Id;
+            Number = secondOpinion.Number;
+            Description = secondOpinion.Description;
+            Title = secondOpinion.Title;
+            CreatedById = secondOpinion.CreatedById;
+            CreatedBy = new QueryPerson(secondOpinion.CreatedBy);
+            CreatedDate = secondOpinion.CreatedDate;
+
+            if(secondOpinion.Responses is not null)
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Domain/Models/SecondOpinions/QuerySecondOpinion.cs:20]: According to the db-model the expression here is always true, so do you need it?

File: src/backend/api/Fusion.Resources.Domain/Models/SecondOpinions/QuerySecondOpinion.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            {
+                Responses = secondOpinion.Responses.Select(x => new QuerySecondOpinionResponse(x));
+            }
+
+            if(secondOpinion.Request is not null)
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Domain/Models/SecondOpinions/QuerySecondOpinion.cs:25]: According to the db-model the expression here is always true, so do you need it?

File: src/backend/api/Fusion.Resources.Api/Controllers/Requests/SecondOpinionController.cs [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+            }
+
+            Response.Headers.Add("Allow", string.Join(',', allowed));
+            return NoContent();
+        }
+
+        [HttpPatch("/resources/requests/internal/{requestId}/second-opinions/{secondOpinionId}/")]
+        public async Task<ActionResult<ApiSecondOpinion>> PatchSecondOpinion(Guid requestId, Guid secondOpinionId, [FromBody] PatchSecondOpinionRequest payload)
+        {
+            var requestItem = await DispatchAsync(new GetResourceAllocationRequestItem(requestId));
```
[the remark this answers is not in the document]
Comment (loldot) [reply src/backend/api/Fusion.Resources.Api/Controllers/Requests/SecondOpinionController.cs:190]: Both request item and second opinion is required in auth check

File: src/backend/api/Fusion.Resources.Domain/Commands/SecondOpinions/GetSecondOpinions.cs (hunk 1 above)
Comment (xssic) [reply src/backend/api/Fusion.Resources.Domain/Commands/SecondOpinions/GetSecondOpinions.cs:112]: Ser fremdeles ikke helt noen god grunn til å endre på filter input

**Review by xssic — COMMENTED** [review]
Did you remember to add aka-config in all environments?

Also. Not exactly sure why you "rewrite" the filter argument where mentioned.
Can't you just make front-end filter on appropriate values from backend?

**Review by xssic — APPROVED** [review]

**Pull request merged**